pkg/snapshot/snapshot.go

```diff
@@ -208,12 +209,18 @@ func (s *Snapshotter) snapShotFS(f io.Writer) (bool, error) {
 		}
 		if maybeAdd {
 			logrus.Debugf("Adding %s to layer, because it was changed.", path)
+			addedPaths = append(addedPaths, path)
 			filesAdded = true
 			if err := t.AddFileToTar(path); err != nil {
 				return false, err
 			}
 		}
 	}
 
+	if filesAdded && len(addedPaths) == 1 && addedPaths[0] == "/" {
+		logrus.Infof("Only added /, not taking snapshot.")
```

Comment:
Why is this change needed?

Reply:
so for a command like `RUN touch /${FOO}`, where $FOO doesn't resolve to anything, there should just be an empty layer appended to the image, that's what docker does. For some reason, we see that / has changed and add that to the snapshot, resulting in an extra non-empty layer compared to docker.

Reply:
Hrmm, what if it's like "/foo/${FOO}"? Maybe the check is if the only added path is a directory, rather than root?

Reply:
yep, done.